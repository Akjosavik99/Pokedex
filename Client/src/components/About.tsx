import { Typography } from "@mui/material";
import { Box } from "@mui/system";

export default function About() {
  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        width: "50vw",
        margin: "auto",
        marginTop: "5vh",
      }}
    >
      <Typography variant="h1">About</Typography>
      <Typography variant="body1">
        This is a Pokedex where you can search for your favorite Pokemon and add
        them to your team.
      </Typography>

      <Typography variant="body1">Hope you like it and have fun!</Typography>
    </Box>
  );
}
