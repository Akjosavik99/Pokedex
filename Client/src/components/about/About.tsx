import { Typography } from "@mui/material";
import { Box } from "@mui/material";
import { Divider } from "@mui/material";
import pokeBall from "../../assets/Pokeball.png";

export default function About() {
  const styles = {
    about: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "space-between",
      width: {
        xs: "90vw",
        sm: "80vw",
        md: "60vw",
      },
      margin: "auto",
      marginTop: "5vh",
      color: "primary.main",
      textAlign: "center",
    },
    paragraph: {
      marginTop: "5vh",
    },
    paragraphAlingment: {
      textAlign: {
        xs: "left",
        sm: "center",
      },
    },
  };

  return (
    <Box sx={styles.about}>
      <Typography variant="h2">About</Typography>
      <Typography sx={styles.paragraphAlingment}>
        Discover our Pokedex, a handy tool for Pokemon enthusiasts featuring
        information on 300 Pokemons from the first and second generations. Dive
        into a world of statistics, where you can delve into details like
        height, weight, type, and more for each Pokemon. Share your own unique
        insights and reviews about these creatures with fellow users. But that's
        not all - you can also curate your dream team of six Pokemons to keep
        them close, offering you a fun and engaging way to stay connected with
        your favorite companions.
      </Typography>

      <Typography variant="h4" sx={styles.paragraph}>
        Search
        <Typography sx={styles.paragraphAlingment}>
          You can search for a pokemon by its name or you can filter pokemons
          based on different types (water, fire, grass etc.). Additionally you
          can sort pokemons by their name, height, weight, and base experience,
          in both ascending and descending order. The results will show you the
          first 20 pokemons that match your search criteria, with the option to
          go to the next page to see 20 more pokemons.
        </Typography>
      </Typography>

      <Typography variant="h4" sx={styles.paragraph}>
        My Team
        <Divider />
        <Typography sx={styles.paragraphAlingment}>
          You have the ability to put together your own team of Pokémon. You can
          add up to 6 Pokémon to your team before it's full. You also have the
          ability to remove Pokémon from your team if you suddenly change your
          mind. And no, you cannot have 6 Charizards.
        </Typography>
      </Typography>

      <Typography
        variant="h4"
        sx={{ marginTop: "5vh", color: "primary.light" }}
      >
        Gotta Catch 'Em All!
      </Typography>
      <img src={pokeBall} id="pokeball" alt="pokeball" />
    </Box>
  );
}
