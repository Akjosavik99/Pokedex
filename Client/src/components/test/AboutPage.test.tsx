import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import About from "../about/About";

describe("About", () => {
  test("Snapshot test of page", () => {
    const page = render(<About />);
    expect(page).toMatchSnapshot();
  });
});
