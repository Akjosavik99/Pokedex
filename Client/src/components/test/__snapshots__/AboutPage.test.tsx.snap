// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About > Snapshot test of page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-2ahs1"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          About
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1j8009j-MuiTypography-root"
          tabindex="0"
        >
          Discover our Pokedex, a handy tool for Pokemon enthusiasts featuring information on 300 Pokemons from the first and second generations. Dive into a world of statistics, where you can delve into details like height, weight, type, and more for each Pokemon. Share your own unique insights and reviews about these creatures with fellow users. But that's not all - you can also curate your dream team of six Pokemons to keep them close, offering you a fun and engaging way to stay connected with your favorite companions.
        </p>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-114bf3r-MuiTypography-root"
        >
          Search
          <p
            class="MuiTypography-root MuiTypography-body1 css-1j8009j-MuiTypography-root"
            tabindex="0"
          >
            You can search for a pokemon by its name or you can filter pokemons based on different types (water, fire, grass etc.). Additionally you can sort pokemons by their name, height, weight, and base experience, in both ascending and descending order. The results will show you the first 20 pokemons that match your search criteria, with the option to go to the next page to see 20 more pokemons.
          </p>
        </h4>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-114bf3r-MuiTypography-root"
        >
          My Team
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-1j8009j-MuiTypography-root"
            tabindex="0"
          >
            You have the ability to put together your own team of Pokémon. You can add up to 6 Pokémon to your team before it's full. You also have the ability to remove Pokémon from your team if you suddenly change your mind. And no, you cannot have 6 Charizards.
          </p>
        </h4>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-xq6mha-MuiTypography-root"
          tabindex="0"
        >
          Gotta Catch 'Em All!
        </h4>
        <img
          alt="pokeball"
          id="pokeball"
          src="/project2/assets/Pokeball.png"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-2ahs1"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        About
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1j8009j-MuiTypography-root"
        tabindex="0"
      >
        Discover our Pokedex, a handy tool for Pokemon enthusiasts featuring information on 300 Pokemons from the first and second generations. Dive into a world of statistics, where you can delve into details like height, weight, type, and more for each Pokemon. Share your own unique insights and reviews about these creatures with fellow users. But that's not all - you can also curate your dream team of six Pokemons to keep them close, offering you a fun and engaging way to stay connected with your favorite companions.
      </p>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-114bf3r-MuiTypography-root"
      >
        Search
        <p
          class="MuiTypography-root MuiTypography-body1 css-1j8009j-MuiTypography-root"
          tabindex="0"
        >
          You can search for a pokemon by its name or you can filter pokemons based on different types (water, fire, grass etc.). Additionally you can sort pokemons by their name, height, weight, and base experience, in both ascending and descending order. The results will show you the first 20 pokemons that match your search criteria, with the option to go to the next page to see 20 more pokemons.
        </p>
      </h4>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-114bf3r-MuiTypography-root"
      >
        My Team
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-1j8009j-MuiTypography-root"
          tabindex="0"
        >
          You have the ability to put together your own team of Pokémon. You can add up to 6 Pokémon to your team before it's full. You also have the ability to remove Pokémon from your team if you suddenly change your mind. And no, you cannot have 6 Charizards.
        </p>
      </h4>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-xq6mha-MuiTypography-root"
        tabindex="0"
      >
        Gotta Catch 'Em All!
      </h4>
      <img
        alt="pokeball"
        id="pokeball"
        src="/project2/assets/Pokeball.png"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
